src/backend/distributed/transaction/transaction_management.c

```diff
@@ -391,6 +392,9 @@ CoordinatedTransactionCallback(XactEvent event, void *arg)
 			ResetGlobalVariables();
 			ResetRelationAccessHash();
 
+			/* Reset any local replication origin session since transaction has been aborted.*/
```

Comment:
Don't we need to `ResetReplicationOriginLocalSession` for commits as well?

Reply:
Replication Origin session can span multiple transactions.. 
We are only resetting for aborts, just as a safety mechanim, in case there is some fatal failure.
We already have ResetReplicationOriginLocalSession  calls in normal flow and MemoryContextReset callback as well..